HashTable/hashtable.c

```diff
@@ -0,0 +1,145 @@
+#include "../List/list.h"
+#include <stdlib.h>
+#include "hashtable.h"
+#include <stdio.h>
+#include <string.h>
+#define TABLE_SIZE 1000
+
+typedef struct HT {
+    Node* elements[TABLE_SIZE];
+} HashTable;
+
+HashTable* initTable(void) {
+    HashTable* table = malloc(sizeof(HashTable));
+    if (table == NULL) {
+        perror("Memory allocate error");
+        return NULL;
+    }
+    for (int i = 0; i < TABLE_SIZE; ++i) {
+        (table->elements)[i] = NULL;
+    }
+    return table;
+}
+
+unsigned int hashFunction(char* word, int length) {
+    int hash = 0;
+    for (int i = 0; i < length ; i++) {
+        char current = word[i];
+        hash += current;
```

Comment:
```suggestion
        hash += (unsigned int)word[i];
```
И это далеко не лучшая хеш-функция для строк